### DISCUSSION TITLE
What is a way to use socks proxy?

### DISCUSSION THREAD

**Discussion by Pooort** [OpenVPN/openvpn3-linux#206] [Q&A - Our support arena]
It seems like `socks-proxy` setting is not supported in client ovpn file.

`Client -- ERROR --: Connection failed: option_error: sorry, unsupported options present in configuration: UNKNOWN/UNSUPPORTED OPTIONS`

How to use socks proxy with openvpn3 client?

**Comment by schwabe** [comment]
The OpenVPN3 library does not implement socks proxy. If you need that feature, you should use OpenVPN 2.x

**Comment by Pooort** [comment]
@schwabe Thank you.

```
** ERROR ** /usr/bin/openvpn2: error: unrecognized arguments: --socks-proxy 127.0.0.1 1080
```
What am I doing wrong?

**Reply by schwabe** [reply]
That is not openvpn 2.x. Check `openvpn --version` to see the version.

**Reply by Pooort** [reply]
@schwabe Thanks a lot.
But I can't use `openvpn` due to: `This profile requires web based SAML authentication`.
It's strange that there's no appropriate client on Linux. On Windows everything works without any issues.

**Reply by schwabe** [reply]
We don't have a UI for web ui of openvpn 2.x on Linux. You can add `--setenv IV_SSO openurl,webauth` to tell the server that you can do web auth and then manually copy the URL.

**Reply by Pooort** [reply]
@schwabe Ok. Almost there. I have A_LINK from client like:
```
2023-09-11 18:20:39 AUTH_PENDING received, extending handshake timeout from 60s to 180s
2023-09-11 18:20:39 Info command was pushed by server ('WEB_AUTH::A_LINK')
2023-09-11 18:20:40 SENT CONTROL [OpenVPN Server]: 'PUSH_REQUEST' (status=1)
```
But after web auth, in a browser I have:
```
SAML authentication failed

No matching VPN session found
```
How to pass the auth?

**Reply by schwabe** [reply]
The URL should work. The error message sounds like the server is OpenVPN Access Server. And it does not care about you copying the url or whatssoever als long as you keep the session in that url in that `A_LINK` intact. Should be something like `vpnauth~nodeid~sessiond` or just `vpnauth~~sessiond` in that URL.

**Reply by Pooort** [reply]
Hm. So strange. Yes the url has:
`RelayState=vpnauth~~PG_H12kIhea7atP5gIx`
The same ovpn file works with `openvpn3` and doesn't work with `openvpn`.

**Reply by schwabe** [reply]
That would really surprise me. That feature has been well tested and as long as you have the full URL it should work. My own app OpenVPN for Android or OpenVPN GUI also implement that feature and generally work okay.

**Reply by Pooort** [reply]
@schwabe Thank you a lot.
It works really. The issue was in url coping.
There's another issue related to the tunnel setting, will investigate by myself now.
Arne, thank you one more time for great support.
